tensorflow/core/kernels/fixed_length_record_reader_op.cc

```diff
@@ -62,14 +64,20 @@ class FixedLengthRecordReader : public ReaderBase {
 
   Status ReadLocked(string* key, string* value, bool* produced,
                     bool* at_end) override {
-    if (input_buffer_->Tell() >= file_pos_limit_) {
+    if (input_buffer_->Tell() >= file_pos_limit_ ||
+        input_buffer_->Tell() + record_bytes_ > file_pos_limit_) {
```

Comment:
It seems to me the 2 cases you described are expected behaviors.  I believe the caller of this reader class is responsible for setting up the reader configuration so that footer bytes are disallowed if desired (and vice versa).  Do you agree?